jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/job/Job.java

```diff
@@ -54,12 +61,39 @@
     public String getName();
 
     /**
-     * Get priority of job. Only 1 = high priority, 2 = medium and 3 = low are allowed.
+     * Get priority of job. Only 1 = high priority, 2 = medium and 3 = low are
+     * allowed.
      * <p>
      * Default is 2 = medium.
      *
      * @return priority
      */
     public int getPriority();
 
+
+    /**
+     * @return All involved locations
+     */
+    public List<Location> getAllLocations();
+
+    /**
+     * @return Start location
+     */
+    public Location getStartLocation();
+
+    /**
+     * @return End location
+     */
+    public Location getEndLocation();
+
+    /**
+     * @return All activities
+     */
+    public JobActivityList getActivityList();
```

Comment:
If we need all time windows, activity durations or locations, we should just iterate JobActivityList and grab the corresponding activity. I think we do not need them often without relation to the correponding activity. What do you think?